x/leverage/keeper/liquidate.go

```diff
@@ -0,0 +1,84 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/umee-network/umee/v2/x/leverage/types"
+)
+
+// computeLiquidation takes a repayment and reward denom proposed by a liquidator and calculates
+// the actual repayment amount a target address is eligible for, and the corresponding collateral
+// to burn and rewards to return to the liquidator.
+func (k Keeper) computeLiquidation(
```

Comment:
Refactor note: This un-exported keeper function handles the parts of liquidation that require a the leverage `Keeper` - mostly fetching balances and parameters. It then calls the purely mathematical function in `types` package.